### DISCUSSION TITLE
status=Bad_Timeout

### DISCUSSION THREAD

**Discussion by yasodakrishnav** [eclipse-milo/milo#1232] [General]
Hi there.
I'm getting below exception while readValues()

Reading n number of nodes --> List<DataValue> dataValues = client.readValues(0.0, TimestampsToReturn.Both, nodeIdList).get();

UaException: status=Bad_Timeout, message=requestId=3 timed out after 60000ms

i tried setting request timeout to 120000ms in OpcUaClientConfigBuilder, still didn't work. is there any property that i can change to work with slow responding opc servers.

Thanks

**Comment by kevinherron** [comment]
You probably need to read less Nodes at once rather than keep increasing the timeout.

**Reply by yasodakrishnav** [reply]
But i am trying to read the values for just 238 tags

**Reply by yasodakrishnav** [reply]
this is happening when i tried to read bulk nodes, if any of the node takes time to read value, its throwing UaException: status=Bad_Timeout exception.

is there any way to skip reading this values if its takes time continue reading the values of other nodes while reading value using readValues() method?

**Reply by kevinherron** [reply]
No, when you call readValues() the Nodes are read with a single call to the Read service. It’s not possible to know which Node(s) are problematic. The entire service call times out.

**Reply by yasodakrishnav** [reply]
is there any difference between readValue() and readValues() methods. if both works same way like reading single node at time. i can you the readValue() method with minimal timeout if its throw timeout exception, i can catch the exception and continue reading other nodes right?

**Reply by kevinherron** [reply]
If you use the `readValue` function you will be reading 1 Node at a time, and reading each Node will have its own timeout. It might take a while or be less efficient to read all your Nodes this way, but it may work.

**Reply by yasodakrishnav** [reply]
can you explain how does readValues() methods in-depth if possible?
and also could you please suggest me better way to do this?

**Reply by kevinherron** [reply]
It calls `OpcUaClient::read`: 
https://github.com/eclipse/milo/blob/cd0f3d97f78de3823541229032642f8af9f00040/opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/OpcUaClient.java#L509-L523

There is no "good" way to do this - just ways that might suck less. You need a more reliable server.

**Reply by yasodakrishnav** [reply]
Alright @kevinherron.. Thanks for you time and help!
